pkg/url/url.go

```diff
@@ -85,39 +86,53 @@ func Get(client *occlient.Client, localConfig *config.LocalConfigInfo, urlName s
 	return URL{}, errors.New(fmt.Sprintf("the url %v does not exist", urlName))
 }
 
-// GetIngress returns ingress spec for given URL name
-func GetIngress(kClient *kclient.Client, envSpecificInfo *envinfo.EnvSpecificInfo, urlName string, componentName string) (URL, error) {
+// GetIngressOrRoute returns ingress/route spec for given URL name
+func GetIngressOrRoute(client *occlient.Client, kClient *kclient.Client, envSpecificInfo *envinfo.EnvSpecificInfo, urlName string, componentName string) (URL, error) {
 	remoteExist := true
 	// Check whether remote already created the ingress
-	ingress, err := kClient.GetIngress(urlName)
-	if kerrors.IsNotFound(err) {
+	ingress, getIngressErr := kClient.GetIngress(urlName)
+	// Check whether remote already created the route
+	route, getRouteErr := client.GetRoute(urlName)
+	if kerrors.IsNotFound(getIngressErr) && kerrors.IsNotFound(getRouteErr) {
 		remoteExist = false
-	} else if err != nil {
-		return URL{}, errors.Wrap(err, "unable to get ingress on cluster")
+	} else if getIngressErr != nil && getRouteErr != nil {
```

Comment:
Shouldn't this be a `or` condition? If either of the them is an actual error, then we should immediately error out rather than proceeding forward.